core/main/src/broker/thunder/user_data_migrator.rs

```diff
@@ -0,0 +1,918 @@
+    config_file_path: String,                   // path to the migration map file
+    response_tx: Sender<BrokerOutput>,
+    response_rx: Arc<Mutex<Receiver<BrokerOutput>>>,
+}
+
+impl UserDataMigrator {
+    pub fn create() -> Option<Self> {
+        let possible_config_file_paths = vec![
+            format!("/etc/{}", USER_DATA_MIGRATION_CONFIG_FILE_NAME),
+            format!(
+                "{}/{}",
+                RIPPLE_STORAGE_DIR, USER_DATA_MIGRATION_CONFIG_FILE_NAME
+            ),
+            format!("./{}", USER_DATA_MIGRATION_CONFIG_FILE_NAME),
+        ];
+
+        for path in possible_config_file_paths {
+            if Path::new(&path).exists() {
+                debug!("Found migration map file: {}", path);
+                if let Some(migration_map) = Self::load_migration_config(&path) {
+                    let (response_tx, response_rx) = mpsc::channel(16);
+                    return Some(UserDataMigrator {
+                        migration_config: Arc::new(Mutex::new(migration_map)),
+                        config_file_path: path.to_string(),
+                        response_tx,
+                        response_rx: Arc::new(Mutex::new(response_rx)),
+                    });
+                }
+            }
+        }
+        debug!("No migration map file found");
+        None
+    }
+
+    async fn get_matching_migration_entry_by_method(
+        &self,
+        method: &str,
+    ) -> Option<MigrationConfigEntry> {
+        let migration_map = self.migration_config.lock().await;
+        migration_map
+            .values()
+            .find(|entry| entry.getter == method || entry.setter == method)
+            .cloned()
+    }
+
+    /// function to intercept and handle broker request. Perform migration if needed
+    pub async fn intercept_broker_request(
+        &self,
+        broker: &ThunderBroker,
+        ws_tx: Arc<Mutex<SplitSink<WebSocketStream<TcpStream>, Message>>>,
+        request: &mut BrokerRequest,
+    ) -> bool {
+        let method = request.rpc.method.clone();
+        info!(
+            "intercept_broker_request: Intercepting broker request for method: {:?}",
+            method
+        );
+        if let Some(config_entry) = self.get_matching_migration_entry_by_method(&method).await {
+            if config_entry.setter == method {
+                return self
+                    .handle_setter_request(broker, ws_tx, request, &config_entry)
+                    .await;
+            } else {
+                return self
+                    .handle_getter_request(broker, ws_tx, request, &config_entry)
+                    .await;
+            }
+        }
+
+        info!(
+            "intercept_broker_request: No migration entry found for method: {:?}",
+            method
+        );
+        // Continue with the original request if no migration entry is found
+        false
+    }
+
+    async fn handle_setter_request(
```

Comment:
`handle_setter_request` will be called from `intercept_broker_request`  for those firebolt set api calls listed in migration config file. 

The use case here is to always update `LegacyStorage` with the **new value**  while continuing with the normal set operation of new plugin. The `LegacyStorage` update is required to avoid data loss in case of rollback.